app/javascript/controllers/embed_preview_controller.js

```diff
@@ -1,141 +1,52 @@
 import { Controller } from "@hotwired/stimulus"
 
 export default class extends Controller {
-  static targets = ["accentColor", "embedIframe", "embedIframeWrapper", "embedUrl", "embedHtml"]
-
-  static values = {
-    embedUrl: String,
-    embedType: String,
-  }
-
-  connect() {
-    const that = this
-    window.addEventListener(
-      "message",
-      (e) => {
-        that.handlePostMessage(e)
-      },
-      false
-    )
-
-    if (this.hasEmbedIframeTarget && this.hasEmbedUrlValue) {
-      this.embedIframeTarget.setAttribute("src", this.embedUrlValue)
-    }
-  }
-
-  disconnect() {
-    const that = this
-    window.removeEventListener(
-      "message",
-      (e) => {
-        that.handlePostMessage(e)
-      },
-      false
-    )
-  }
-
-  handlePostMessage(e) {
-    // Bail if post message didn't originate from a Play domain.
-    if (!/^https?:\/\/play(?:\.staging)?\.prx\.(?:org|tech|test)$/.test(e.origin)) return
-
-    this.previewSource = e.source
-    this.previewOrigin = e.origin
-
-    this.updateEmbedUrlTarget(e.data.embedUrl)
-    this.updateEmbedUHtmlTarget(e.data.embedHtml)
-    this.updateEmbedStyles(e.data.embedStyles, e.data.embedHeight)
-  }
-
-  postConfigChange(config) {
-    this.previewSource.postMessage(config, this.previewOrigin)
-  }
-
-  updateEmbedUrlTarget(embedUrl) {
-    if (!embedUrl) return
-
-    for (const elm of this.embedUrlTargets) {
-      switch (elm.tagName) {
-        case "INPUT":
-          elm.value = embedUrl
-          break
-
-        case "A":
-          elm.href = embedUrl
-          break
-
-        case "BUTTON":
-          elm.setAttribute("data-clipboard-copy-value", embedUrl)
-          break
+  static targets = ["wrapper"]
+  static values = { embedHtml: String }
+
+  // handle changes to iframe html ourselves, to prevent flickering
+  embedHtmlValueChanged(value, previousValue) {
+    if (previousValue) {
+      const wrapper = this.wrapperTarget
+      const iframe = this.wrapperTarget.querySelector("iframe")
+      const tpl = document.createElement("template")
+      tpl.innerHTML = value
+
+      if (tpl.content.firstChild.tagName === "IFRAME") {
+        this.updateWrapper(wrapper, null)
+        this.updateIframe(iframe, tpl.content.firstChild)
+      } else {
+        this.updateWrapper(wrapper, tpl.content.firstChild)
+        this.updateIframe(iframe, tpl.content.firstChild.firstChild)
```

Comment:
Yeah, the idea here is the string `embedHtmlValue` changes, and its this controllers responsibility to make those changes to the `wrapperTarget` div containing the iframe.

It seemed easiest to turn the string into an Element, and interrogate that for what styles/attributes we should change.

Yeah, there could be a cleaner way to do this?  I thought maybe I could call [morphdom](https://github.com/patrick-steele-idem/morphdom) manually here, and it would basically do all this for me?  But couldn't quite figure it out.